src/libs/actions/Session/index.js

```diff
@@ -424,20 +424,29 @@ function validateEmail(accountID, validateCode, password) {
         });
 }
 
-// It's necessary to throttle requests to reauthenticate since calling this multiple times will cause Pusher to
-// reconnect each time when we only need to reconnect once. This way, if an authToken is expired and we try to
+// We only need to reconnect once. If an authToken is expired, and we try to
 // subscribe to a bunch of channels at once we will only reauthenticate and force reconnect Pusher once.
-const reauthenticatePusher = _.throttle(() => {
+let isReconnectingPusher = false;
+function reauthenticatePusher() {
+    if (isReconnectingPusher) {
+        return;
+    }
+
+    isReconnectingPusher = true;
     Log.info('[Pusher] Re-authenticating and then reconnecting');
+
     API.reauthenticate('Push_Authenticate')
         .then(Pusher.reconnect)
         .catch(() => {
             console.debug(
                 '[PusherConnectionManager]',
                 'Unable to re-authenticate Pusher because we are offline.',
             );
+        })
+        .finally(() => {
+            isReconnectingPusher = false;
         });
-}, 5000, {trailing: false});
+}
```

Comment:
I noticed this was still causing extra Push Authenticate calls 

![Multiple Authenticate and Push Authenticate requests](https://user-images.githubusercontent.com/12156624/142208919-9efaba54-e300-4782-bd7c-9ccc85ae1bff.png)

Reply:
I don't know what you think you are looking at. But anytime we subscribe to a channel a call to `Push_Authenticate` is made.

Reply:
This opening/focusing App after more than 2 hours are passed since last usage (expired token)

Reply:
This logic as well as the throttle might be unnecessary anymore
Depends on what we want to achieve 

Since `reauthenticate` would hook to the same authentication call it means that we won't be triggering multiple authentication calls anymore 

But the calls would still end up calling `Pusher.reconnect` and that might be undesired 

I prefer this logic over `throttle` as it guarantees a single call

Since this change (and the unit tests for it) are a bit out of scope I can move it  to a separate PR

Reply:
@johnmlee101 dropped this change from the PR
I've opened a new PR here: #6707